src/main.test.ts

```diff
@@ -0,0 +1,433 @@
+      actor: 'workflowactor',
+      serverUrl: 'https://github.com',
+    };
+
+    // All tests will work under the AUT directory
+    process.env.GITHUB_WORKSPACE = join(process.cwd(), 'aut');
+
+    mockCore.getInput.mockImplementation(
+      (name: string): string => mock.inputs[name] || '',
+    );
+
+    mockCache.restoreCache.mockImplementation(
+      (
+        paths: string[],
+        primaryKey: string,
+        restoreKeys?: string[] | undefined,
+        options?: DownloadOptions | undefined,
+      ): Promise<string | undefined> =>
+        new Promise((resolve) => {
+          if (fs.existsSync('last-run-status')) {
+            resolve('test');
+          } else {
+            resolve(undefined);
+          }
+        }),
+    );
+
+    // Setting to this Github repo by default
+    (github.context.payload.repository as Partial<PayloadRepository>) = {
+      clone_url: mock.repo,
+    };
+    (github.context.workflow as string) = mock.workflow;
+    (github.context.runId as number) = mock.runId;
+    (github.context.payload.pull_request as Partial<PayloadRepository>) = {
+      head: {
+        ref: mock.headRef,
+      },
+    };
+    (github.context.ref as string) = mock.ref;
+    (github.context.eventName as string) = mock.eventName;
+    (github.context.actor as string) = mock.actor;
+    (github.context.serverUrl as string) = mock.serverUrl;
+  });
+
+  beforeAll(() => {
+    fs.writeFileSync('last-run-status', `completed/failure`, {
+      encoding: 'utf8',
+    });
+  });
+
+  afterAll(() => {
+    delete process.env.GITHUB_WORKSPACE;
+    fs.unlink('last-run-status', (err) => {
+      if (err) {
+        console.error(err);
+      }
+    });
+  });
+
+  it('should send success notification if last run failed and current succeed', async () => {
+    await run();
+    expect(mockCore.setFailed).toHaveBeenCalledTimes(0);
+    expect(mockCore.info).toHaveBeenCalledWith('Current run status: success');
+    expect(mockCore.info).toHaveBeenCalledWith('Success notification');
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Last run status: completed/failure',
+    );
+    expect(mockCore.info).toHaveBeenCalledWith(
+      'Message body: {"username":"workflow-status-slack-notification CI alert","icon_emoji":":bangbang:","attachments":[{"color":"good","author_name":"workflowactor","author_link":"https://github.com/workflowactor","author_icon":"https://github.com/workflowactor.png?size=32","fields":[{"title":"Repository","value":"workflow-status-slack-notification","short":true},{"title":"Branch","value":"main","short":true},{"title":"Action URL","value":"<https://github.com/reside-eng/workflow-status-slack-notification/actions/runs/23456|Publish Action>","short":true},{"title":"Event","value":"pull_request","short":true},{"title":"Publish Action workflow success","value":"Previously failing Publish Action workflow in workflow-status-slack-notification succeed.","short":false}]}]}',
+    );
```

Comment:
Might be good to create an inline snapshot for this to make it easier to maintain. I used them recently and found them of great benefit.

Reply:
Thanks a lot, I'm going to check what is an inline snapshot and will improve this part of the tests